opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -128,6 +128,12 @@ func resourceOpennebulaVirtualMachine() *schema.Resource {
 				Computed:    true,
 				Description: "Current LCM state of the VM",
 			},
+			"poweroff_hard_on_resize": {
```

Comment:
I'm asking myself for this attributes if we shouldn't make something more generic.
The name is too specific, and we may want the do the same thing in other places ? (For now only delete action use the `terminate-hard` action)

i.e. make it effective for all RPC calls that poweroff/terminate/reboot the VMs ([see available hard actions: ](https://docs.opennebula.io/5.12/integration/system_interfaces/api.html#one-vm-action)).

The `-hard` action should be used when the VM doesn’t have ACPI support.
[see 5.12 documentation](https://docs.opennebula.io/6.2/management_and_operations/vm_management/vm_instances.html#basic-virtual-machine-operations)

We may want a name like `power_hard`, `hard_action` ... ?
An idea @frousselet ?

Reply:
I agree.

Maybe something like `action_on_resize` (type String instead of Boolean)?
Example: `action_on_resize = "poweroff-hard"`

Reply:
I'm not sure you understood what I mean,
I meant this attribute could be used through the whole VM resource, not only for resize feature. 
This would only switch on the hard behavior, not the action type (i.e. poweroff/terminate/reboot).
For instance, for delete action this could also be used to switch between hard/regular behavior

Am I clear ?

Reply:
Not sure, we only want to power off the VM to resize the VM.
I think it would be better ton only be able to switch the hard behavior for the whole provider. This would also be useful when deleting. aVM where a terminate action is used

Reply:
I am not sure to understand.
What do you mean by 'whole provider'?

Please don't hesitate to provide HCL code examples.

Reply:
To be more accurate I meant for the whole virtual machine resource.

VM actions that have a `-hard` equivalent are: `terminate`, `undeploy`, `poweroff`, and `reboot`.
Currently for the VM resource we only use the `terminate-hard` action to delete the VM,
and this PR suggest to use`poweroff` and optionally the `-hard` equivalent action.

Here's what I propose: to use actions `terminate`, `undeploy`, `poweroff`, and `reboot` by default, then add a flag that allow to switch to the `-hard` behavior.

The doc mention that the `-hard` behavior should be used when the VM doesn’t have ACPI support, so we could also name it like `acpi_support`, `hard_shutdown`